.github/workflows/build-and-test-bridge.yml

```diff
@@ -66,13 +65,49 @@ jobs:
             wasm/tx_*.wasm
             wasm/vp_*.wasm
             wasm/checksums.json
+
+  test-wasm:
+    timeout-minutes: 30
+    runs-on: ${{ matrix.os }}
+    needs: [build-wasm]
+    container: 
+      image: ghcr.io/anoma/namada:wasm-0.8.0
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        wasm_cache_version: ["v2"]
+        mold_version: [1.7.0]
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        if: ${{ github.event_name != 'pull_request_target' }}
+      - name: Checkout PR
+        uses: actions/checkout@v3
+        if: ${{ github.event_name == 'pull_request_target' }}
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Duplicate checksums file
+        run: cp wasm/checksums.json wasm/original-checksums.json
+      - name: Install mold linker
+        run: |
+          wget -q -O- https://github.com/rui314/mold/releases/download/v${{ matrix.mold_version }}/mold-${{ matrix.mold_version }}-x86_64-linux.tar.gz | tar -xz
+          mv mold-${{ matrix.mold_version }}-x86_64-linux/bin/mold  /usr/local/bin
+      - name: Download wasm artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: wasm-${{ github.event.pull_request.head.sha|| github.sha }}
+          path: ./wasm
+        env:
+          RUSTFLAGS: "-C linker=clang -C link-arg=-fuse-ld=/usr/local/bin/mold"
       - name: Test Wasm
         run: make test-wasm
       - name: Check wasm up-to-date
-        run: cmp -- wasm/checksums.json wasm/original-checksums.json
+        run: cmp -- wasm/checksums.json wasm/original-checksums.json || true
```

Comment:
will this still catch `if: failure()` below?

Reply:
oh my bad here, the bridge team asked me to let the CI pass when if the checksum is not up-to-date. But this is the wrong workflow file, ill fix this! Thanks